### PR TITLE
Initial commit of code snippets for measurements.

### PR DESCRIPTION
**Pull request by deflaux**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# How to contribute a snippet
```
Comment (calbach) [inline README.md:9]: I'm not sure which section this goes in, but could you put some developer setup instructions somewhere in here which mention setting up git-secrets, e.g. a copy-paste or link to this? https://github.com/all-of-us/workbench#git-secrets

We also have a git hook to ensure this is installed, but I can send a separate PR for that.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Don't like these conventions? We can change them! This is just a starting point. Keep in mind we'll need to reflect those changes in the auto-generation script described in the next section.
+
+# Auto-generation of Jupyter 'Snippets Menu' configuration.
+
+For more detail, see the [Snippets Menu](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/snippets_menu/readme.html) Jupyter extension documentation.
+
+1. Generate the JSON configuration:
+    ```
+    git clone https://github.com/all-of-us/code-snippets.git
+    cd code-snippets/build
```
Comment (calbach) [inline README.md:32]: nit: should there be a virtualenv + pip install step here?

**Review by calbach — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# How to contribute a snippet
+
+1. Write the SQL to retrieve the data of interest.
+    * Try to put as much of the data wrangling logic into SQL as possible. This means there's less data wrangling code to write twice (e.g., once in [pandas](https://pandas.pydata.org/) for Python and again [dplyr](https://dplyr.tidyverse.org/) for R).
+    * Choose a good prefix for your `<my-query>.sql` filename. The file name helps users decide whether its useful to them. It will also become the name of the dataframe holding the query results (e.g. `<my-query>_df`).
+    * Put some comments in your query too. Look at the other `.sql` files for examples.
+    * Tip: Paste your SQL into the [BigQuery web UI](https://bigquery.cloud.google.com/) and click on the 'format' button to format it nicely.
```
Comment (kelseyMayo) [inline README.md:15]: @deflaux given the impending changes to the workbench security boundary that will prevent users from using the BQ UI, should we propose alternatives that do not use the BQ UI?

File: snippets/measurement_of_interest_by_age_and_gender.ggplot [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# This plot assumes that most_recent_measurement_of_interest.sql has been run.
+options(repr.plot.height = 14, repr.plot.width = 8)
+
+measurement_of_interest_df %>%
+    filter(value_as_number < 9999999) %>% # Get rid of nonsensical outliers.
```
Comment (kelseyMayo) [inline snippets/measurement_of_interest_by_age_and_gender.ggplot:5]: @deflaux just food for thought, for these code snippets should we  consider adding additional data cleaning steps like removal of NANs prior to plotting?

File: snippets/snippets_setup.R [hunk 5]
```diff
@@ -0,0 +1,32 @@
+library(viridis)    # A nice color scheme for plots.
+library(ggthemes)   # Common themes to change the look and feel of plots.
+library(scales)     # Graphical scales map data to aesthetics in plots.
+library(skimr)      # Better summaries of data.
+library(lubridate)  # Date library from the tidyverse.
+library(bigrquery)  # BigQuery R client.
+library(tidyverse)  # Data wrangling packages.
+
+## CHANGE THESE AS NEEDED - default parameter values for snippets.
+DATASET <- 'aou-res-curation-output-prod.R2019Q1R2'
```
Comment (kelseyMayo) [inline snippets/snippets_setup.R:10]: @deflaux @calbach theoretically, even at launch users will have the ability to access multiple CDRs for analysis. At a minimum they will have access to a synthetic CDR and a real CDR. They may also have access to older CDRs for the purposes of replication. How can we design snippet setup such that it defaults to the CDR version selected at workspace setup?

**Review by kelseyMayo — COMMENTED** [review]
@deflaux I had a few questions that I left in comments below. Overall this is a great foundation for our code snippet work. Thank you again for your work on this!

File: README.md (hunk 1 above)
Comment (deflaux) [reply README.md:9]: Added a new section at the top with this.

File: README.md (hunk 2 above)
Comment (deflaux) [reply README.md:32]: Added a comment about the pip command.

File: README.md (hunk 3 above)
Comment (deflaux) [reply README.md:15]: No need! The BigQuery UI can still be a great tool in our toolbox. Let's just provide world-readable OMOP tables with a few rows of fake data for testing purposes. Kinda like the fake data in the SQL unit tests.

See https://github.com/deflaux/code-snippets/blob/master/snippets/measurements_of_interest_summary_test.py#L50

File: snippets/measurement_of_interest_by_age_and_gender.ggplot (hunk 4 above)
Comment (deflaux) [reply snippets/measurement_of_interest_by_age_and_gender.ggplot:5]: Agreed, good food for thought!!

When passed null data, conveniently ggplot2 and plotnine drop it and spit out a warning to bring to your attention that this has occurred.

Other needed cleaning will depend upon the particular measurement. So I added one cleaning step that should(?) generally be fine and hopefully that bit of code that prompts people to realize that they can add more logic right there if they want to change the plot further by omitting/modifying more of the data.

File: snippets/snippets_setup.R (hunk 5 above)
Comment (deflaux) [reply snippets/snippets_setup.R:10]: Its a great question and I don't have an answer for it right now that I like. Can you please file an issue on this repo for that and lets keep thinking about it?

For now its not too hard for people to edit it manually, and for us to update the checked-in snippet when there is a new release.

**Comment by deflaux** [comment]
Thanks for the feedback @kelseyMayo and @calbach !

File: README.md (hunk 3 above)
Comment (kelseyMayo) [reply README.md:15]: Sounds good!

File: snippets/measurement_of_interest_by_age_and_gender.ggplot (hunk 4 above)
Comment (kelseyMayo) [reply snippets/measurement_of_interest_by_age_and_gender.ggplot:5]: Excellent - that addresses my basic question. Thanks!

File: snippets/snippets_setup.R (hunk 5 above)
Comment (calbach) [reply snippets/snippets_setup.R:10]: Agreed - we discussed this last week as well. We don't have an easy fix for it so it probably wouldn't be part of the v1.

**Review by calbach — APPROVED** [review]

File: snippets/snippets_setup.R (hunk 5 above)
Comment (kelseyMayo) [reply snippets/snippets_setup.R:10]: Will do.

**Review by kelseyMayo — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**